server/src/main/java/org/elasticsearch/client/internal/ClusterAdminClient.java

```diff
@@ -222,6 +225,25 @@
      */
     NodesInfoRequestBuilder prepareNodesInfo(String... nodesIds);
 
+    /**
+     * Cluster formation info.
+     *
+     * @return The result future
+     * @see org.elasticsearch.client.internal.Requests#clusterFormationInfoRequest(String...)
+     */
+    ActionFuture<ClusterFormationInfoResponse> clusterFormationInfo(ClusterFormationInfoRequest request);
```

Comment:
I normally wouldn't bother adding dedicated methods to the client for an API that's not going to be generally useful. You can just call `Client#execute` directly in the few places where it's needed.